### ISSUE TITLE
combine queries to speed up loading

### ISSUE THREAD

**Issue by berger-n** [buda-base/public-digital-library#941]
for example on https://library-dev.bdrc.io/show/bdr:MW22084
- the data about the instance: 
  https://ldspdi-dev.bdrc.io/query/graph/ResInfo-SameAs?R_RES=bdr%3AMW22084&format=json
- the outline: 
  http://ldspdi-dev.bdrc.io/query/graph/Outline_root?R_RES=bdr%3AMW22084
- the data about the scans: 
  http://ldspdi-dev.bdrc.io/query/graph/ResInfo-SameAs?R_RES=bdr%3AW22084&format=json
- the data about the work: 
  http://ldspdi-dev.bdrc.io/query/graph/ResInfo-SameAs?R_RES=bdr%3AWA0BC001&format=json
- the data about the etext: 
  http://ldspdi-dev.bdrc.io/query/graph/ResInfo-SameAs?R_RES=bdr%3AIE0OPI6440489A&format=json
- the etext outline: 
  http://ldspdi-dev.bdrc.io/query/graph/etextrefs?R_RES=bdr%3AIE0OPI6440489A
- more data about the etext: 
  http://ldspdi-dev.bdrc.io/query/graph/Etext_base?I_LIM=500&R_RES=bdr%3AVLIE0OPI6440489A_I0886&format=json
- the first pages of the etext: 
  http://ldspdi-dev.bdrc.io/osearch/etextchunks?cstart=0&cend=10000&id=bdr%3AVLIE0OPI6440489A_I0886
- the iiif collection: 
  http://iiifpres.bdrc.io/collection/wio:bdr:MW22084::bdr:W22084
- the iiif manifest:
  https://iiifpres.bdrc.io/vo:bdr:I0886/manifest
- even more iiif data:
  http://iiifpres.bdrc.io/il/v:bdr:I0886

**Comment by eroux** [comment]
Thanks! Let's start with a query that combines instance, scan and work. It should also have enough data to make another quick query for the etext snippet, without having to call the query for the etext outline.

I think it's fine if the outline stays in a different query for now. But maybe the website can detect that the user is on a MW and just start the query of the outline before it gets the return of the main query?

**Comment by berger-n** [comment]
> maybe the website can detect that the user is on a MW and just start the query of the outline before it gets the return of the main query

not so sure about this... this was a long time ago but see for example https://github.com/buda-base/public-digital-library/issues/641 or the commit in https://github.com/buda-base/public-digital-library/issues/754 
(there are several cases, it's not a one-size-fits-all query)

**Comment by eroux** [comment]
thanks! that's much faster, although still some room for improvement for the etext... the best for that would probably be to have a dedicated endpoint for the snippet, I'll try to do that next week

**Comment by eroux** [comment]
Thanks! Now I'm actually wondering why the collection and manifest are queried, we have the URL of the thumbnail from the query right, we could just display it?

**Comment by berger-n** [comment]
note that in the case of https://library-dev.bdrc.io/show/bdr:MW22084_0044_2
there's an additional call to [`ResInfo_MW`](https://ldspdi-dev.bdrc.io/query/graph/ResInfo-SameAs_MW?R_RES=bdr%3AMW22084&format=json) (regarding the root instance)
which maybe could be merged in the first one?

**Comment by eroux** [comment]
ah yes, much better, thanks!

Let's merge also the call to the root MW. 

I'm not really persuaded we need to have an option to diplay the images with the etext in the preview, it seems a bit too much for the preview... what do you think @roopeux ?

I think in terms of UX, the preview of the etext should be bound vertically to a relatively short thing that users can scroll, now it takes too much space I think

Yes I kind of remember complicated cases for the manifest... but maybe we can find a way to improve it too, I think we've reached a pretty good state here so it's maybe less of a priority, but at some point can you list the uses of the manifest so we can see if we can replace it?

**Comment by eroux** [comment]
P.S.: can you tell me what you need from the root instance?

**Comment by roopeux** [comment]
I agree, the etext preview should be visually about the same size as the scans.
I don't think it even adds much if users can scroll the preview.  It could be pretty short and click on anywhere in that div would open the etext viewer.

So if displaying images from etext preview introduces any kind of tech complication it should be disabled.

**Comment by berger-n** [comment]
an attempt at a shorter, non scrollable, fully clickable etext preview:

![image](https://github.com/user-attachments/assets/b6f58974-3534-4846-b645-592e8996d4d4)

---

the last example is still a bit heavy to load (because of the outline queries) but the others are now quite fast (here there's no outline, or it is loaded only when asked):
- https://library-dev.bdrc.io/show/bdr:MW1KG22580
- https://library-dev.bdrc.io/show/bdr:MW22084_0047
- https://library-dev.bdrc.io/show/bdr:MW22084

---

regarding the iiif manifests, having a look at https://library.bdrc.io/show/bdr:MW22084_0045 reminds me that we used to find the first image of a "subinstance" from them (which has not been reimplemented yet in the new UX, though it should be straightforward using the data that's already there)

but could there be this dedicated thumbnail like in https://ldspdi-dev.bdrc.io/query/graph/ResInfo-SameAs_MW?R_RES=bdr%3AMW22084_0045&format=json? 

and also the scan info is not in the new query, but is shown after loading the outline/root query (for example here: https://library-dev.bdrc.io/show/bdr:MW22084_0051)

**Comment by eroux** [comment]
that's really good, this is a great improvement!
